Source/URLRequestConvertible.swift

```diff
@@ -42,36 +42,119 @@ extension URLRequest: URLRequestConvertible {
 }
 
 extension URLRequest {
+    
+    @available(*, deprecated, message: "Use the new initializer using [String:String]? as the parameters value")
+    init(path: String, baseURL: URL,
+                HTTPMethod: HTTPMethod = .GET, parameters: [String: Any]? = nil,
+                body: Data? = nil, allHTTPHeaderFields: Dictionary<String, String>? = nil) {
+        let mappedParameters: [String:String]?
+        if let parameters = parameters {
+            var convertedParams = [String: String]()
+            for (key, value) in parameters {
+                convertedParams[key] = "\(value)"
+            }
+            mappedParameters = convertedParams
+        } else {
+            mappedParameters = nil
+        }
```

Comment:
I would do:
```swift
let mappedParameters = parameters.map {
var convertedParams = [String: String]()
            var convertedParams = [String: String]()
            for (key, value) in parameters {
                convertedParams[key] = "\(value)"
             }
        return convertedParams
}
```

To avoid the if let thing.

But more important this is not tested. Could lead to a gegression